### PR TITLE
[WIP] Guin decorating the space

### PR DESCRIPTION
**Pull request by guineveresaenger**
Walked through the instructions today and putting up this PR ISO feedback

- pulled in the correct provider dependency from jfrog (finally!)
- Successfully ran `make tfgen` and noted warnings
- Mapped Resources and Data Sources
- Added jfrog as the dependent org
- Ran make provider
    - Encountered a couple of the following warnings that I need to read up on but am hoping they refer to example code snippets:
    ```
    warning: Unexpected code snippets in section # Artifactory Remote Repository Resource for resource 
    artifactory_remote_repository
    ```
- Ran make build_sdks
- Seem to have successfully generated most SDKs but dotnet wants to be special:
```
Build FAILED.

/Users/guin/go/src/github.com/pulumi/pulumi-artifactory/sdk/dotnet/AccessToken.cs(24,31): error CS0542: 'AccessToken': member names cannot be the same as their enclosing type [/Users/guin/go/src/github.com/pulumi/pulumi-artifactory/sdk/dotnet/Pulumi.Artifactory.csproj]
/Users/guin/go/src/github.com/pulumi/pulumi-artifactory/sdk/dotnet/ApiKey.cs(28,31): error CS0542: 'ApiKey': member names cannot be the same as their enclosing type [/Users/guin/go/src/github.com/pulumi/pulumi-artifactory/sdk/dotnet/Pulumi.Artifactory.csproj]
    0 Warning(s)
    2 Error(s)
```
- Successfully ran go mod tidy on the go sdk
- Attempted to run `lint_provider` and discovered that `golangci-lint` is a dependency that I do not have.

TL;DR: the dotnet SDK is unhappy about a namespace issue, and I don't know where my linter's at. Will pick this up tomorrow. :D 

Fixes: #2

### CODE REVIEW

**Comment by jkodroff** [comment]
@guineveresaenger The DotNet error is a thing that comes up semi-regularly.  You need to add an override to `resources.go` so that the namespace/class name isn't e.g. `Foo.Foo` in C#, like: https://github.com/pulumi/pulumi-aws/blob/master/provider/resources.go#L334-L340

File: provider/go.mod [hunk 1]
```diff
@@ -8,7 +8,9 @@ replace (
 )
 
 require (
-	github.com/hashicorp/terraform-plugin-sdk v1.9.1
+	github.com/hashicorp/terraform-plugin-sdk v1.9.1 // indirect
+	github.com/jfrog/terraform-provider-artifactory v0.0.0-20211108203830-a482006c47c5
 	github.com/pulumi/pulumi-terraform-bridge/v3 v3.9.0
-	github.com/pulumi/pulumi/sdk/v3 v3.14.0
+	github.com/pulumi/pulumi/sdk/v3 v3.14.1-0.20211007222624-789e39219452
```
Comment (jkodroff) [inline provider/go.mod:14]: Should be `v3.17.0` (latest Pulumi version).

File: provider/go.mod [hunk 2]
```diff
@@ -8,7 +8,9 @@ replace (
 )
 
 require (
-	github.com/hashicorp/terraform-plugin-sdk v1.9.1
+	github.com/hashicorp/terraform-plugin-sdk v1.9.1 // indirect
+	github.com/jfrog/terraform-provider-artifactory v0.0.0-20211108203830-a482006c47c5
 	github.com/pulumi/pulumi-terraform-bridge/v3 v3.9.0
```
Comment (jkodroff) [inline provider/go.mod:13]: This should be changed to `v3.11.0` (latest version).

**Review by jkodroff — CHANGES_REQUESTED** [review]

File: provider/go.mod (hunk 1 above)
Comment (guineveresaenger) [reply provider/go.mod]: Done, thanks to you :)

File: provider/go.mod (hunk 2 above)
Comment (jkodroff) [inline provider/go.mod]: Missed this because it's not changed.  This should also be 3.11.0.

**Comment by guineveresaenger** [comment]
I added workflows and opened the corresponding `ci-mgmt` PR: https://github.com/pulumi/ci-mgmt/pull/87

I am not entirely certain which Secrets to set since we don't have Artifactory set up in the Actions environment (maybe there's some magic I don't understand yet 😅 )

I couldn't figure out the Python package dependency so I filed #2 as well as #3, so as to not get stuck.

I did run `make test` locally and the one test in the `examples` directory passes.

Any advice on what to put into the CHANGELOG would be welcome also.

**Review by stack72 — APPROVED** [review]
